coldfront/core/project/views.py

```diff
@@ -902,15 +912,33 @@ def post(self, request, *args, **kwargs):
 
             project_user_update_form = ProjectUserUpdateForm(request.POST,
                                                              initial={'role': project_user_obj.role.name,
-                                                                      'enable_notifications': project_user_obj.enable_notifications}
-                                                             )
+                                                                      'enable_notifications': project_user_obj.enable_notifications})
 
             if project_user_update_form.is_valid():
                 form_data = project_user_update_form.cleaned_data
-                project_user_obj.enable_notifications = form_data.get(
-                    'enable_notifications')
-                project_user_obj.role = ProjectUserRoleChoice.objects.get(
-                    name=form_data.get('role'))
+                project_user_obj.enable_notifications = form_data.get('enable_notifications')
+
+                old_role = project_user_obj.role
+                new_role = ProjectUserRoleChoice.objects.get(name=form_data.get('role'))
+                only_manager = not project_obj.projectuser_set.filter(~Q(pk=project_user_pk),
+                                                                      role__name__in=['Manager']).exists()
+
+                # the only manager of this project, is demoting himself
+                if old_role.name == 'Manager' and new_role.name != 'Manager' and only_manager:
+                    project_pis = project_obj.projectuser_set.filter(role__name__in=['Principal Investigator'])
+
+                    if not project_pis.exists():
+                        # no pis exist, cannot demote
+                        old_role = new_role
+                        messages.info(request, 'There are no PIs on this project, cannot step down from Manager role.')
+                    else:
+                        # update all pis to receive notifications
+                        messages.info(request, 'Stepping down from Manager role, all PIs now receive notifications.')
```

Comment:
Let's change this to `messages.warning`.

Again in the case when the last Manager is being demoted (not promoted), let's change the wording to: "User {username} is no longer a manager. All PIs will now receive notifications.".

(On a related note, I don't recall if we decided: (a) all PIs should have their notifications enabled, or (b) a manager would select one PI to receive them before being able to stop being a manager".)

Reply:
I dont think we discussed (b), and I remember bringing up (a) during the weekly meetings. Lets sync up on Monday before closing this.